### ISSUE TITLE
Image.show() problem with I;16 mode TIFFs

### ISSUE THREAD

**Issue by Felix-Droop** [python-pillow/Pillow#4729]
### What did you do?

Loaded a 16 bit integer TIFF file and showed it.

### What did you expect to happen?

That the image looks exactly the same as when opened manually.

### What actually happened?

It looked very different. 

I can't share the original image I am working with, but it becomes almost completely white, whereas it was mostly black before. 

I generated another example (code below). I can't upload it here, because GitHub doesn't accept .tif. 

The black one on the left side is the actual correct one opened manually.

![Screenshot ](https://user-images.githubusercontent.com/62176388/85736642-f2d1b700-b6fe-11ea-9528-e196d0e89138.png)

 The problem might be the temporary conversion to BMP. (Uneducated guess)

### What are your OS, Python and Pillow versions?

* OS: Windows 10 and WSL Ubuntu 18.04
* Python: 3.8.2
* Pillow: 7.1.2

```python
from PIL import Image 

# code used to generate the example image
# img = Image.new('I;16', (1000,1000), 70)

# code that causes issue
with Image.open('test_generated.tif') as img:
    print(img.mode) # I;16
    print(img.format) # TIFF 
    img.show()
```

EDIT:
[test_generated.zip](https://github.com/python-pillow/Pillow/files/4836715/test_generated.zip)

**Comment by radarhere** [comment]
If you compress your original image into a zip or tgz, then you should be able to upload it.

**Comment by wiredfool** [comment]
I'm pretty sure it's because the default conversions are using the 0-255 range of the i;16 and clipping the rest to 255.

**Comment by Felix-Droop** [comment]
> If you compress your original image into a zip or tgz, then you should be able to upload it.

I'm not allowed to share it. The generated example is enough to reproduce and analyze the error, isn't it?
I uploaded it as a .zip.

> I'm pretty sure it's because the default conversions are using the 0-255 range of the i;16 and clipping the rest to 255.

When generating the example picture (code above) I used higher number than 255 for the intensitiy (instead of now 70). They all produced blank white images. Makes sense regarding your explanation.

**Comment by radarhere** [comment]
I was just reading your message that you couldn't upload a .tif, and trying to offer a solution.

Regarding your problem, it's something of a known issue. See #3159, and the related #3011

**Comment by nulano** [comment]
Note that #4080 was merged and released in 7.1.0, meaning that `show` should now always convert the image to PNG, not BMP. However, due to the following line, images are always converted to a different mode unless they are in one of 1, RGBA, or LA modes: https://github.com/python-pillow/Pillow/blob/4f6145655b6fa35ec8dd3600041418087e399758/src/PIL/ImageShow.py#L62
This is likely no longer necessary, since PNG supports I;16 mode, as well as most (if not all) others.

**Comment by radarhere** [comment]
Thanks @nulano. I've created PR #4744 accordingly.
